www/js/lib/uiUtil.js

```diff
@@ -74,13 +74,16 @@ define([], function() {
     }
         
     var regexpRemoveUrlParameters = new RegExp(/([^\?]+)\?.*$/);
+    var regexpRemoveAnchor = new RegExp(/([^#]+)#.*$/);
     
+    /**
+     * Removes parameters and anchors from a URL
+     * @param {type} url
+     * @returns {String} same URL without its parameters and anchors
+     */
     function removeUrlParameters(url) {
-        if (regexpRemoveUrlParameters.test(url)) {
-            return regexpRemoveUrlParameters.exec(url)[1];
-        } else {
-            return url;
-        }
+        var urlWithoutParameters = regexpRemoveUrlParameters.test(url) ? url.match(regexpRemoveUrlParameters)[1] : url;
+        return regexpRemoveAnchor.test(urlWithoutParameters) ? urlWithoutParameters.match(regexpRemoveAnchor)[1] : urlWithoutParameters;
```

Comment:
@mossroy This all works fine, but it's unnecessarily verbose. You can just do:
```
function removeUrlParameters(url) {
        return url.replace(regexpRemoveUrlParameters, "$1");
}
```
Alternatively, there's no real need to precompile such a simple regex. So you could get rid of the two regexp definitions above this function and here you can just do:
```
function removeUrlParameters(url) {
        return url.replace(/([^?#]+)[?#].*$/, "$1");
}
```
I tested this on Firefox, Edge and FFOS, and it has exactly the same effect as yours. I understand that for readability you may prefer the version with the precompiled regex.

Reply:
I agree. I'll push a commit to use your much better method.
Regarding precompiling the regexp, I did not do a benchmark but I suppose it's worth it as you avoid to recompile it on each function call.